src/main/java/io/curity/identityserver/plugin/webauthn/authenticate/WebAuthnSelectDeviceRequestHandler.java

```diff
@@ -52,16 +52,21 @@
     private final WebAuthnPluginConfiguration _configuration;
     private final SessionManager _sessionManager;
     private final AccountManager _accountManager;
-    private final UserPreferenceManager _userPreferenceManager;
     private final ExceptionFactory _exceptionFactory;
+    private final String _usernameInAuthSession;
 
     public WebAuthnSelectDeviceRequestHandler(WebAuthnPluginConfiguration configuration)
     {
         _configuration = configuration;
         _accountManager = configuration.getAccountManager();
         _sessionManager = configuration.getSessionManager();
-        _userPreferenceManager = configuration.getUserPreferenceManager();
         _exceptionFactory = configuration.getExceptionFactory();
+
+        WebAuthnAuthenticationSession webAuthnAuthenticationSession = WebAuthnAuthenticationSession.readFromSession(
```

Comment:
This call should not be done in the constructor of a request handler.
The request handler might be created but not used , in which case it should not cause any side-effects.
Please move this call to the handler method where it's needed.

Reply:
Alright. I have moved the session retrieval to the preProcess method. This way I can use in both get() and post() handlers.